File: Data Structures & Algorithms/palindromic-substrings/submission-2.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        n = len(s)
        if n <= 1:
            return n
        dp = [[False] * n for _ in range(n)] 
        count = 0
        for i in range(n):
            dp[i][i] = True
            count += 1
        for i in range(n - 1):
            if s[i] == s[i+1]:
                dp[i][i+1] = True
                count += 1
        for sub_str_len in range(3, n + 1):
            for i in range(n - sub_str_len + 1):
                j = i + sub_str_len - 1
                if s[i] == s[j] and dp[i+1][j-1]:
                    dp[i][j] = True
                    count += 1
        return count

File: Data Structures & Algorithms/meeting-schedule/submission-2.cs
/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

public class Solution {
    public bool CanAttendMeetings(List<Interval> intervals) {
        var n  = intervals.Count;
        if (n == 0)
        {
            return true;
        }
        intervals = intervals.OrderBy(x => x.start).ToList();
        var prevEnd = intervals[0].end;
        for (var i = 1; i < n; i++)
        {
            var interval = intervals[i];
            if (prevEnd > interval.start)
            {
                return false;
            }
            else 
            {
                prevEnd = interval.end;
            }
        }
        return true;
    }
}
